grafana/provider.go

```diff
@@ -344,6 +385,16 @@ func createSMClient(d *schema.ResourceData) (string, *smapi.Client) {
 	return smURL, smapi.NewClient(smURL, smToken, nil)
 }
 
+func createAmixrClient(d *schema.ResourceData) (*amixrAPI.Client, error) {
+	aToken := d.Get("amixr_access_token").(string)
+	base_url := d.Get("amixr_url").(string)
+	aclient, err := amixrAPI.New(base_url, aToken)
+	if err != nil {
+		return nil, err
+	}
+	return aclient, nil
+}
```

Comment:
```suggestion
	return amixrAPI.New(base_url, aToken)
}
```
You can return immediately, the client will be ignored if there's an err